src/plugins/data_source_management/public/plugin.ts

```diff
@@ -91,7 +94,10 @@ export class DataSourceManagementPlugin
       registerAuthenticationMethod(sigV4AuthMethod);
     }
 
-    return { registerAuthenticationMethod };
+    return {
+      registerAuthenticationMethod,
+      getDataSourcePicker: createClusterSelector(),
```

Comment:
So if I am understanding correctly, this will expose a function `getDataSourcePicker`, that plugins should be able to consume via `optionalPlugins`?

Reply:
Exactly!